### PR TITLE
#1590 Part 1 -  Ministry confirm COE

### PR DESCRIPTION
**Pull request by dheepak-aot**
## Ministry Confirm COE - Part 1 of #1590 

- [x] Add "confirm enrolment" button on ministry disbursement summary page (mockup 3)
image

**Note:** Confirmed from @JasonCTang that ministry should be able confirm enrolments even outside the COE window(21 days).

- [x] If user clicks confirm enrolment display yes/no modal
image

image

- [x] Tuition remittance always = 0 if completed by ministry
- [x] Add role to keycloak AEST user as "student-confirm-enrolment" add to groups: aest-business-administrators

image
image

- [x] Modified the get COE Summary Query to get disbursement schedules which belong to assessments which are current assessments of of applications.

### CODE REVIEW

File: sources/packages/backend/apps/api/src/constants/error-code.constants.ts [hunk 1]
```diff
@@ -121,3 +121,12 @@ export const DUPLICATE_INSTITUTION_LOCATION_CODE =
  * Duplicate SABC code for the institution.
  */
 export const DUPLICATE_SABC_CODE = "DUPLICATE_SABC_CODE";
+
+export const COE_NOT_FOUND_MESSAGE =
```
Comment (ann-aot) [inline sources/packages/backend/apps/api/src/constants/error-code.constants.ts:125]: this is an error message, right? but the file name is `error-code.constants`.

File: sources/packages/backend/apps/api/src/route-controllers/confirmation-of-enrollment/confirmation-of-enrollment.controller.service.ts [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+      !this.applicationService.withinValidCOEWindow(
+        new Date(disbursementSchedule.disbursementDate),
+      )
+    ) {
+      throw new UnprocessableEntityException(
+        `Confirmation of Enrollment window is greater than ${COE_WINDOW} days`,
+      );
+    }
+
+    // TODO: Add validation for past study period.
```
Comment (andrepestana-aot) [inline sources/packages/backend/apps/api/src/route-controllers/confirmation-of-enrollment/confirmation-of-enrollment.controller.service.ts:70]: Are you going to work on this `TODO` ?

File: sources/packages/backend/apps/api/src/route-controllers/confirmation-of-enrollment/confirmation-of-enrollment.aest.controller.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+import { COE_NOT_FOUND_MESSAGE } from "../../constants";
+import { ClientTypeBaseRoute } from "../../types";
+import { COE_WINDOW } from "../../utilities";
+import BaseController from "../BaseController";
+import { ConfirmationOfEnrollmentControllerService } from "./confirmation-of-enrollment.controller.service";
+
+@AllowAuthorizedParty(AuthorizedParties.aest)
+@Groups(UserGroups.AESTUser)
+@Controller("confirmation-of-enrollment")
+@ApiTags(`${ClientTypeBaseRoute.AEST}[Confirmation Of Enrollment - CoE]`)
```
Comment (ann-aot) [inline sources/packages/backend/apps/api/src/route-controllers/confirmation-of-enrollment/confirmation-of-enrollment.aest.controller.ts:26]: we were following this format in all other controllers, 
image

File: sources/packages/backend/apps/api/src/route-controllers/confirmation-of-enrollment/confirmation-of-enrollment.aest.controller.ts [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+   * An application can have up to two COEs based on the disbursement schedule,
+   * hence the COE approval happens twice for application with more than once disbursement.
+   * Irrespective of number of COEs to be approved, application status is set to complete
+   * on first COE approval.
+   * @param disbursementScheduleId disbursement schedule id of COE.
+   */
+  @ApiNotFoundResponse({ description: COE_NOT_FOUND_MESSAGE })
+  @ApiUnprocessableEntityResponse({
+    description:
+      `Confirmation of enrollment window is greater than ${COE_WINDOW} days ` +
```
Comment (ann-aot) [inline sources/packages/backend/apps/api/src/route-controllers/confirmation-of-enrollment/confirmation-of-enrollment.aest.controller.ts:45]: mm.. I think I heard there would be no 21 day check for AEST users right?  Am I missing something

Comment (ann-aot) [reply sources/packages/backend/apps/api/src/route-controllers/confirmation-of-enrollment/confirmation-of-enrollment.aest.controller.ts:45]: pls review the description

File: sources/packages/web/src/components/common/ConfirmEnrolment.vue [hunk 5]
```diff
@@ -0,0 +1,84 @@
+<template>
+  <check-permission-role :role="Role.StudentConfirmEnrolment">
+    <template #="{ notAllowed }">
+      <v-btn
+        v-if="isConfirmCOEEnabled"
```
Comment (ann-aot) [inline sources/packages/web/src/components/common/ConfirmEnrolment.vue:5]: what is the differnce between `isConfirmCOEEnabled` and `notAllowed` here?

File: sources/packages/web/src/components/common/ConfirmEnrolment.vue [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+        variant="text"
+        color="primary"
+        @click="submitConfirmEnrolment"
+        :disabled="notAllowed"
```
Comment (ann-aot) [inline sources/packages/web/src/components/common/ConfirmEnrolment.vue:9]: if an AEST does not have `StudentConfirmEnrolment` role, what do they see `Confirm Enrolment` btn/link, or its hidden for them?

File: sources/packages/web/src/components/common/ConfirmEnrolment.vue [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+        ><span class="text-decoration-underline"
+          ><strong>Confirm Enrolment</strong></span
```
Comment (ann-aot) [inline sources/packages/web/src/components/common/ConfirmEnrolment.vue:11]: for bold pls use the style class that uses `$font-bold;` added by @hellolynn-tbtb

File: sources/packages/web/src/services/ConfirmationOfEnrollmentService.ts [hunk 8]
```diff
@@ -4,7 +4,7 @@ import {
   ApplicationDetailsForCOEAPIOutDTO,
   COEDeniedReasonAPIOutDTO,
   COESummaryAPIOutDTO,
-  ConfirmationOfEnrollmentAPIInDTO,
+  ConfirmEnrollmentOptions,
```
Comment (ann-aot) [inline sources/packages/web/src/services/ConfirmationOfEnrollmentService.ts:7]: why `APIInDTO,` is removed?

File: sources/packages/web/src/services/http/dto/ConfirmationOfEnrolment.dto.ts [hunk 9]
```diff
@@ -16,6 +16,17 @@ export interface COESummaryAPIOutDTO {
...
 
+export interface ConfirmEnrollmentOptions {
+  /**
+   * location id of the application.
+   */
+  locationId?: number;
+  /**
+   * COE confirmation information.
+   */
+  payload?: ConfirmationOfEnrollmentAPIInDTO;
```
Comment (ann-aot) [inline sources/packages/web/src/services/http/dto/ConfirmationOfEnrolment.dto.ts:27]: got it

File: sources/packages/web/src/components/common/AssessmentAwardDetails.vue [hunk 10]
```diff
@@ -182,15 +204,25 @@ import { COEStatus, StatusInfo } from "@/types";
...
+    ConfirmEnrolment,
+    StatusInfoEnrolment,
+  },
   props: {
     assessmentAwardData: {
       type: Object as PropType<AwardDetailsAPIOutDTO>,
       required: true,
       default: {} as AwardDetailsAPIOutDTO,
     },
+    allowConfirmEnrolment: {
```
Comment (ann-aot) [inline sources/packages/web/src/components/common/AssessmentAwardDetails.vue:222]: what is this?

File: sources/packages/web/src/views/aest/student/AssessmentAward.vue [hunk 11]
```diff
@@ -59,14 +66,19 @@ export default defineComponent({
     const assessmentAwardData = ref<AwardDetailsAPIOutDTO>();
     const { mapAssessmentDetailHeader } = useAssessment();
     const headerMap = ref<Record<string, string>>({});
+    const snackBar = useSnackBar();
 
     onMounted(async () => {
+      loadAssessmentAwardValues();
```
Comment (ann-aot) [inline sources/packages/web/src/views/aest/student/AssessmentAward.vue:72]: `onMounted(loadAssessmentAwardValues)`

File: sources/packages/backend/apps/api/src/constants/error-code.constants.ts (hunk 1 above)
Comment (andrewsignori-aot) [reply sources/packages/backend/apps/api/src/constants/error-code.constants.ts:125]: Most of the time I would say that the error names are constants and the error messages aren't, with few exceptions like these ones. I am not sure what was the motivation to have these ones as constants in the past.

File: sources/packages/web/src/views/aest/student/AssessmentAward.vue [hunk 12]
```diff
@@ -79,11 +91,30 @@ export default defineComponent({
       });
     };
 
+    const confirmEnrolment = async (disbursementId: number) => {
+      try {
+        await ConfirmationOfEnrollmentService.shared.confirmEnrollment(
+          disbursementId,
+        );
+        snackBar.success("Confirmation of Enrollment Confirmed!");
+        loadAssessmentAwardValues();
```
Comment (ann-aot) [inline sources/packages/web/src/views/aest/student/AssessmentAward.vue:100]: await?

File: sources/packages/web/src/services/http/ConfirmationOfEnrollmentApi.ts [hunk 13]
```diff
@@ -54,21 +54,22 @@ export class ConfirmationOfEnrollmentApi extends HttpBaseClient {
...
-   * @param locationId location id of the application.
    * @param disbursementScheduleId disbursement schedule id of COE.
-   * @param payload COE confirmation information.
+   * @param options Confirm COE options.
    */
   async confirmEnrollment(
-    locationId: number,
     disbursementScheduleId: number,
-    payload: ConfirmationOfEnrollmentAPIInDTO,
+    options?: ConfirmEnrollmentOptions,
```
Comment (andrepestana-aot) [inline sources/packages/web/src/services/http/ConfirmationOfEnrollmentApi.ts:62]: Shouldn't it be an `APIInDTO`?

**Review by andrepestana-aot — COMMENTED** [review]
Great job. Left some comments.

File: sources/packages/web/src/components/common/AssessmentAwardDetails.vue (hunk 10 above)
Comment (ann-aot) [reply sources/packages/web/src/components/common/AssessmentAwardDetails.vue:222]: got it

File: sources/packages/web/src/views/aest/student/AssessmentAward.vue (hunk 11 above)
Comment (andrepestana-aot) [reply sources/packages/web/src/views/aest/student/AssessmentAward.vue:72]: `loadAssessmentAwardValues` is already awaiting, so I think the `async` can be removed as Ann mentioned.

File: sources/packages/backend/apps/api/src/route-controllers/confirmation-of-enrollment/confirmation-of-enrollment.aest.controller.ts (hunk 3 above)
Comment (andrewsignori-aot) [reply sources/packages/backend/apps/api/src/route-controllers/confirmation-of-enrollment/confirmation-of-enrollment.aest.controller.ts:26]: @ann-aot, I believe that @dheepak-aot was trying to follow the below one that we have for locations.
```ts
@ApiTags(`${ClientTypeBaseRoute.Institution}-location[Confirmation Of Enrollment - CoE]`)
```
For the locations, it was added to create the idea of a subitem of the location.
I do not see this as the same for the AEST. I would recommend to use the below.
```ts
@ApiTags(`${ClientTypeBaseRoute.Institution}-Confirmation Of Enrollment - CoE`,)
```
or
```ts
@ApiTags(`${ClientTypeBaseRoute.Institution}-confirmation-of-enrollment`,)
```
and change the location one also to
```ts
@ApiTags(`${ClientTypeBaseRoute.Institution}-location[confirmation-of-enrollment]`)
```

File: sources/packages/web/src/views/aest/student/AssessmentAward.vue [hunk 14]
```diff
@@ -79,11 +91,30 @@ export default defineComponent({
...
+      } catch (error: unknown) {
+        if (
+          error instanceof ApiProcessError &&
+          error.errorType === FIRST_COE_NOT_COMPLETE
+        ) {
+          snackBar.error("First COE is not completed.");
+        }
+        snackBar.error("An error happened while confirming the COE.");
```
Comment (andrepestana-aot) [inline sources/packages/web/src/views/aest/student/AssessmentAward.vue:108]: Doesn't it require an else here?

**Review by andrepestana-aot — COMMENTED** [review]
Great job. Left some comments.

File: sources/packages/backend/apps/api/src/route-controllers/confirmation-of-enrollment/confirmation-of-enrollment.controller.service.ts [hunk 15]
```diff
@@ -0,0 +1,146 @@
...
+
+    const firstOutstandingDisbursement =
+      await this.disbursementScheduleService.getFirstDisbursementSchedule({
```
Comment (andrewsignori-aot) [inline sources/packages/backend/apps/api/src/route-controllers/confirmation-of-enrollment/confirmation-of-enrollment.controller.service.ts:73]: @dheepak-aot let's review the method `getFirstDisbursementSchedule`.
I am a bit confused about why we are trying to get the first pending COE but we are passing the `disbursementSchedule.id`, is it not supposed to be something like the assessment id or application id? Besides that, I guess that this method also needs to be adapted to use the application's current assessment.
Lets review it and decide if any change is needed and if we will take care of this or some other effort.

File: sources/packages/backend/apps/api/src/route-controllers/confirmation-of-enrollment/confirmation-of-enrollment.controller.service.ts (hunk 2 above)
Comment (andrewsignori-aot) [reply sources/packages/backend/apps/api/src/route-controllers/confirmation-of-enrollment/confirmation-of-enrollment.controller.service.ts:70]: It is part of the current ticket probably to be resolved in the upcoming PR.
This PR does not contain the complete implementation of the task.

File: sources/packages/backend/apps/api/src/route-controllers/confirmation-of-enrollment/models/confirmation-of-enrollment.dto.ts [hunk 16]
```diff
@@ -65,3 +65,22 @@ export class COESummaryAPIOutDTO {
   disbursementScheduleId: number;
   disbursementDate: string;
 }
+
+export interface ConfirmEnrollmentOptions {
```
Comment (andrewsignori-aot) [inline sources/packages/backend/apps/api/src/route-controllers/confirmation-of-enrollment/models/confirmation-of-enrollment.dto.ts:69]: Open for discussion but this one maybe should be moved to a `route-controllers/confirmation-of-enrollment/models/confirmation-of-enrollment.models.ts`

File: sources/packages/web/src/components/common/ConfirmEnrolment.vue [hunk 17]
```diff
@@ -0,0 +1,84 @@
...
+import { ModalDialog } from "@/composables";
+import ConfirmModal from "@/components/common/modals/ConfirmModal.vue";
+import CheckPermissionRole from "@/components/generic/CheckPermissionRole.vue";
+
+export default defineComponent({
+  emits: ["confirmEnrolment"],
+  components: { ConfirmModal, CheckPermissionRole },
+  props: {
+    coeStatus: {
+      type: Object as PropType<COEStatus>,
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/components/common/ConfirmEnrolment.vue:47]: If the enums are declared as `type: String as PropType<COEStatus>` they will be typed and Vue will not generate warnings. If I am not wrong, every place that we declared as `Object` it is generating a warning.

File: sources/packages/web/src/components/common/ConfirmEnrolment.vue [hunk 18]
```diff
@@ -0,0 +1,84 @@
...
+
+<script lang="ts">
+import { PropType, computed, defineComponent, ref } from "vue";
+import { COEStatus, Role, ApplicationStatus } from "@/types";
+import { ModalDialog } from "@/composables";
+import ConfirmModal from "@/components/common/modals/ConfirmModal.vue";
+import CheckPermissionRole from "@/components/generic/CheckPermissionRole.vue";
+
+export default defineComponent({
+  emits: ["confirmEnrolment"],
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/components/common/ConfirmEnrolment.vue:43]: Events can be declared as below to have the arguments in a more explicitly way.
```ts
emits: {
    submitted: (submission: unknown, form: FormIOForm) => {
      return !!submission && !!form;
    },
    changed: (form: FormIOForm, event: FormIOChangeEvent) => {
      return !!form && !!event;
    },
    loaded: (form: FormIOForm) => {
      return !!form;
    },
    customEvent: (form: FormIOForm, event: FormIOCustomEvent) => {
      return !!form && !!event;
    },
    render: (form: FormIOForm, event: HTMLElement) => {
      return !!form && !!event;
    },
  },
```

File: sources/packages/web/src/services/ConfirmationOfEnrollmentService.ts (hunk 8 above)
Comment (andrewsignori-aot) [reply sources/packages/web/src/services/ConfirmationOfEnrollmentService.ts:7]: It is confusing having this is the DTO context when it is not a DTO. For the API I recommended moving it to a `models` file and here we can move also, IMO.

File: sources/packages/web/src/services/http/ConfirmationOfEnrollmentApi.ts [hunk 19]
```diff
@@ -54,21 +54,22 @@ export class ConfirmationOfEnrollmentApi extends HttpBaseClient {
...
+      ? `location/${options.locationId}/`
+      : "";
     try {
       await this.patchCall(
         this.addClientRoot(
-          `location/${locationId}/confirmation-of-enrollment/disbursement-schedule/${disbursementScheduleId}/confirm`,
+          `${baseUrl}confirmation-of-enrollment/disbursement-schedule/${disbursementScheduleId}/confirm`,
         ),
-        payload,
+        options?.payload,
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/services/http/ConfirmationOfEnrollmentApi.ts:72]: IMO it is confusing having the payload (that is a DTO) in the options.

File: sources/packages/web/src/views/aest/student/AssessmentAward.vue (hunk 12 above)
Comment (andrewsignori-aot) [inline sources/packages/web/src/views/aest/student/AssessmentAward.vue:100]: Please add the `await`.

**Review by andrewsignori-aot — COMMENTED** [review]
Nice work, please take a look at the comments.

File: sources/packages/web/src/services/http/ConfirmationOfEnrollmentApi.ts (hunk 19 above)
Comment (ann-aot) [reply sources/packages/web/src/services/http/ConfirmationOfEnrollmentApi.ts:72]: I also have similar thought as @andrewsignori-aot

File: sources/packages/backend/apps/api/src/route-controllers/confirmation-of-enrollment/confirmation-of-enrollment.controller.service.ts (hunk 2 above)
Comment (dheepak-aot) [reply sources/packages/backend/apps/api/src/route-controllers/confirmation-of-enrollment/confirmation-of-enrollment.controller.service.ts:70]: @andrepestana-aot  This PR is about Ministry Confirming Enrolment as mentioned in PR description and this todo is about adding validation for past study period which will be an upcoming PR for #1590 .

File: sources/packages/web/src/components/common/ConfirmEnrolment.vue (hunk 6 above)
Comment (dheepak-aot) [reply sources/packages/web/src/components/common/ConfirmEnrolment.vue:9]: I didn't get the question.

File: sources/packages/web/src/components/common/ConfirmEnrolment.vue (hunk 7 above)
Comment (dheepak-aot) [reply sources/packages/web/src/components/common/ConfirmEnrolment.vue:11]: This one ? 
image

File: sources/packages/web/src/views/aest/student/AssessmentAward.vue (hunk 14 above)
Comment (dheepak-aot) [reply sources/packages/web/src/views/aest/student/AssessmentAward.vue:108]: Yes you are right. I can do a return right after line 104.

File: sources/packages/web/src/components/common/ConfirmEnrolment.vue (hunk 7 above)
Comment (dheepak-aot) [reply sources/packages/web/src/components/common/ConfirmEnrolment.vue:11]: I was trying to create a class which just sets the font-family to $font-bold and $font-weight to $font-weight-bold.

The name of the above class was very accurate for that purpose and atomic. I recommend that this class `bold-text` must have been more atomic excluding color definitely and not sure about other properties. Also the variable $font-bold may have been `$font-family-bold` to be more specific.

Coming back to this PR, I created an atomic class `app-font-bold` to set font-weight and font-font family for bold texts.

image

Comment (dheepak-aot) [reply sources/packages/web/src/components/common/ConfirmEnrolment.vue]: Created class font-bold to use for bold only purpose. 

image

File: sources/packages/backend/apps/api/src/services/disbursement-schedule-service/disbursement-schedule-service.ts [hunk 20]
```diff
@@ -382,11 +391,10 @@ export class DisbursementScheduleService extends RecordDataModelService<Disburse
    * only records with coeStatus defined as 'Required' will be considered.
    * @returns first disbursement schedule, if any.
    */
-  async getFirstDisbursementSchedule(options: {
-    disbursementScheduleId?: number;
-    applicationId?: number;
-    onlyPendingCOE?: boolean;
-  }): Promise<DisbursementSchedule> {
+  async getFirstDisbursementScheduleByApplication(
```
Comment (andrewsignori-aot) [inline sources/packages/backend/apps/api/src/services/disbursement-schedule-service/disbursement-schedule-service.ts:394]: How do we ensure that the disbursement belongs to the application's current assessment?
I would say that we can start the query by the application and use the `application.currentAssessment` to return the `DisbursementSchedule`.

Comment (dheepak-aot) [reply sources/packages/backend/apps/api/src/services/disbursement-schedule-service/disbursement-schedule-service.ts:394]: Yes that makes sense and seems to be a great idea to get the disbursements which go through COE process.

File: sources/packages/backend/apps/api/src/constants/error-code.constants.ts [hunk 21]
```diff
@@ -121,3 +121,9 @@ export const DUPLICATE_INSTITUTION_LOCATION_CODE =
  * Duplicate SABC code for the institution.
  */
 export const DUPLICATE_SABC_CODE = "DUPLICATE_SABC_CODE";
+
+export const FIRST_COE_NOT_COMPLETE_MESSAGE =
+  "First disbursement(COE) not complete. Please complete the first disbursement.";
```
Comment (ann-aot) [inline sources/packages/backend/apps/api/src/constants/error-code.constants.ts:126]: so are we planning to keep these error message here?

File: sources/packages/backend/apps/api/src/constants/error-code.constants.ts [hunk 22]
```diff
@@ -121,3 +121,9 @@ export const DUPLICATE_INSTITUTION_LOCATION_CODE =
  * Duplicate SABC code for the institution.
  */
 export const DUPLICATE_SABC_CODE = "DUPLICATE_SABC_CODE";
+
+export const FIRST_COE_NOT_COMPLETE_MESSAGE =
```
Comment (ann-aot) [inline sources/packages/backend/apps/api/src/constants/error-code.constants.ts:125]: so are we planning to keep these error messages here?
```
export const FIRST_COE_NOT_COMPLETE_MESSAGE =

export const INVALID_TUITION_REMITTANCE_AMOUNT_MESSAGE =
```

File: sources/packages/backend/apps/api/src/route-controllers/application/application.students.controller.ts [hunk 23]
```diff
@@ -111,9 +111,9 @@ export class ApplicationStudentsController extends BaseController {
         application.data,
       );
     const firstCOEPromise =
-      this.disbursementScheduleService.getFirstDisbursementSchedule({
+      this.disbursementScheduleService.getFirstDisbursementScheduleByApplication(
```
Comment (ann-aot) [inline sources/packages/backend/apps/api/src/route-controllers/application/application.students.controller.ts:114]: await?

File: sources/packages/web/src/components/common/ConfirmEnrolment.vue [hunk 24]
```diff
@@ -0,0 +1,88 @@
...
+    </template>
+  </check-permission-role>
+
+  <confirm-modal
+    title="Confirm enrolment"
+    ref="confirmEnrolmentModal"
+    okLabel="Confirm enrolment now"
+    cancelLabel="Cancel"
+    ><template #content
+      ><strong
```
Comment (ann-aot) [inline sources/packages/web/src/components/common/ConfirmEnrolment.vue:23]: `bold-text`?

File: sources/packages/web/src/components/common/ConfirmEnrolment.vue [hunk 25]
```diff
@@ -0,0 +1,88 @@
...
+    },
+  },
+  components: { ConfirmModal, CheckPermissionRole },
+  props: {
+    coeStatus: {
+      type: String as PropType<COEStatus>,
+      required: true,
+    },
+    applicationStatus: {
+      type: Object as PropType<ApplicationStatus>,
```
Comment (ann-aot) [inline sources/packages/web/src/components/common/ConfirmEnrolment.vue:55]: `String as`

File: sources/packages/backend/apps/api/src/route-controllers/application/application.students.controller.ts (hunk 23 above)
Comment (dheepak-aot) [reply sources/packages/backend/apps/api/src/route-controllers/application/application.students.controller.ts:114]: It is awaited in Promise,all

image

File: sources/packages/backend/apps/api/src/constants/error-code.constants.ts (hunk 22 above)
Comment (dheepak-aot) [reply sources/packages/backend/apps/api/src/constants/error-code.constants.ts:125]: Forgot to remove those constant. Will remove.

File: sources/packages/backend/apps/api/src/services/disbursement-schedule-service/disbursement-schedule-service.ts (hunk 20 above)
Comment (andrewsignori-aot) [reply sources/packages/backend/apps/api/src/services/disbursement-schedule-service/disbursement-schedule-service.ts:394]: As we talked the COE status must come always from the application's current assessment/reassessment and the COE will be always confirmed by the institution no matter the "Original Assessment" or any reassessment, as per mentioned by @JasonCTang 
@ann-aot @guru-aot @andrepestana-aot FYI

**Review by ann-aot — APPROVED** [review]
Good work @dheepak-aot 👍 . Thanks for doing the changes

**Review by andrepestana-aot — APPROVED** [review]
👍

File: sources/packages/web/src/services/http/ConfirmationOfEnrollmentApi.ts (hunk 19 above)
Comment (andrewsignori-aot) [reply sources/packages/web/src/services/http/ConfirmationOfEnrollmentApi.ts]: @ann-aot the change is ok IMO, hence I am resolving this comment. Please reopen if you have further questions.

**Review by andrewsignori-aot — APPROVED** [review]
Thanks for doing the changes.

**Pull request merged**